### ISSUE TITLE
Add links to prompt to show references

### ISSUE THREAD

**Issue by ishaan1995** [embedchain/embedchain#467]
### 🚀 The feature

Is there a plan to provide links from which embedded chunk was given?
It would make the response better as we could reference and show the links in response as well.
I saw internally in embeddings_queue we do store the link in metadata

### Motivation, pitch

I am creating a bot for content written from writers and want to show original posts in the bot response.

**Comment by cachho** [comment]
including meta_data is definitely on the roadmap, in more capabilities than this.

I can't speak for an ETA.

**Comment by cachho** [comment]
If anyone implements this, please add a template variable and make it configurable as such.

**Comment by denisj** [comment]
> If anyone implements this, please add a template variable and make it configurable as such.

@cachho Why do you mean by template variable?

Could it be something like query and query_with_metadata?

**Comment by smach** [comment]
> Would love to try this framework out, but this is currently a deal breaker for me.

Agree! Users need the ability to see original document chunks  in order to check for accuracy (as well as possibly learn more about their question). Embedchain looks very useful , but I won't deploy RAG apps internally or for the general public unless users can view retrieved source document chunks.

**Comment by deven298** [comment]
> > Would love to try this framework out, but this is currently a deal breaker for me.
> 
> Agree! Users need the ability to see original document chunks in order to check for accuracy (as well as possibly learn more about their question). Embedchain looks very useful , but I won't deploy RAG apps internally or for the general public unless users can view retrieved source document chunks.

Hi @smach, we provide a way to the users to retrieve source document chunks along with answer for `query/chat` method. Please refer to our docs to learn how to do that - https://docs.embedchain.ai/api-reference/app/query

You can use the `citations` flag to retrieve the retrieved source document chunks. Please feel free to reach out to us if you have more questions.

**Comment by DrDavidL** [comment]
On a Chat with PDF page of my repo, I use EmbedChain and process the source text from citations into a readable format for respones! Embedchain is wonderful and feel free to use my code if helpful! https://github.com/DrDavidL/family-chat

**Comment by DrDavidL** [comment]
Also - my code works reliably to return citations for PDF content queries with source text included, which is great! However, this doesn't work for websites added to the database... Thoughts? Same code (just using app.add(site, data_type='web_page')). Queries are answered correctly but citations return empty despite querying with Citations = True. Thoughts? If you can verify databases assembled with data_type = "web_page" do work to return context, I'll recreate my code from scratch to see if I somehow missed something!

**Comment by parshvadaftari** [comment]
Closing as stale
